src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/DlqMessagesToDatabaseTest.java

```diff
@@ -0,0 +1,116 @@
+package uk.gov.hmcts.reform.wacaseeventhandler.controllers;
+
+import io.restassured.response.Response;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.wacaseeventhandler.MessagingTests;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.ccd.message.EventInformation;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.model.CaseEventMessage;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.model.EventMessageQueryResponse;
+import uk.gov.hmcts.reform.wacaseeventhandler.entity.MessageState;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static junit.framework.TestCase.assertEquals;
+import static net.serenitybdd.rest.SerenityRest.given;
+import static org.awaitility.Awaitility.await;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@Slf4j
+@ActiveProfiles(profiles = {"local", "functional"})
+public class DlqMessagesToDatabaseTest extends MessagingTests {
+
+    @Test
+    public void should_store_dlq_messages_in_database()  {
+        List<String> messageIds = List.of(randomMessageId(), randomMessageId(), randomMessageId());
+
+        var caseId = "1111222233334444";
```

Comment:
I would generate the caseId, so we don't run into multiple services hitting AAT failing tests issue.
`var caseId = RandomStringUtils.randomNumeric(16);`

Reply:
Fixed in [a256fa9](https://github.com/hmcts/wa-case-event-handler/pull/224/commits/a256fa9440cf9d1927c7a4d3ee194fcb5f351adb)